From 908c3f377fbcc5beeb217991527c635d2fcc2b66 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Mon, 15 Feb 2021 15:09:38 +0200
Subject: [PATCH 73/76] xen: env: Update env to assign ttyLF0 to Xen and hvc0
 to Dom0

Updated the u-boot environment used in a Xen-enabled setup to
assign LINFlexD_0 (alias serial0) to Xen via bootargs, and to
make Dom0 use hvc0 as console.

Issue: ALB-5666
Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
---
 include/configs/s32.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 4afd20ecb8..23af903902 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -370,14 +370,15 @@
 #endif
 #define XEN_EXTRA_ENV_SETTINGS \
 	"dom0_addr=0x90000000\0" \
-	"bootargs=dom0_mem=192M bootscrub=0\0" \
+	"bootargs=dom0_mem=192M bootscrub=0 console=dtuart dtuart=serial0\0" \
 	"updatexenfdt=fdt addr ${fdt_addr} 0x40000; fdt resize; fdt chosen; " \
 		"fdt set /chosen \\\\\#address-cells <1>; " \
 		"fdt set /chosen \\\\\#size-cells <1>; " \
 		"fdt mknod /chosen module@0; " \
 		"fdt set /chosen/module@0 compatible \"xen,linux-zimage\" \"xen,multiboot-module\"; " \
 		"fdt set /chosen/module@0 reg <${dom0_addr} 0x${filesize} >; " \
-		"fdt set /chosen/module@0 bootargs \"console=ttyLF0,115200 root=/dev/mmcblk0p2 rootwait rw\" \0" \
+		"fdt set /chosen/module@0 bootargs \"console=hvc0 " \
+			"root=/dev/mmcblk0p2 rootwait rw\" \0" \
 	"bootcmd=" XEN_LOAD_FILES "run updatexenfdt; fdt rm /chosen stdout-path; " \
 		"fdt rm /chosen linux,initrd-start; fdt rm /chosen linux,initrd-end; " \
 		"booti ${loadaddr} - ${fdt_addr}\0"
-- 
2.17.1

